File: packages/typescript/src/crs/utmZoneFor.ts
export function utmZoneFor(lng: number): number {
  if (lng >= -84 && lng < -78) {
    return 17;
  }
  if (lng >= -78 && lng < -72) {
    return 18;
  }
  if (lng >= -72 && lng < -66) {
    return 19;
  }
  return Math.floor((lng + 180) / 6) + 1;
}

File: packages/go/utm_zone_for.go
package perspatial

import "math"

func UtmZoneFor(lng float64) int {
	for _, z := range utmCfg.Zones {
		if lng >= z.LngMin && lng < z.LngMax {
			return z.Zone
		}
	}
	return int(math.Floor((lng+180)/6)) + 1
}
